dvc/remote/gdrive.py

```diff
@@ -38,13 +38,37 @@ def __init__(self, path):
         )
 
 
-def gdrive_retry(func):
-    retry_re = re.compile(r"HttpError (403|500|502|503|504)")
+def append_error_details(exc):
+    import json
+    from pydrive2.files import ApiRequestError
+
+    assert isinstance(exc, ApiRequestError)
+
+    if hasattr(exc, "error"):
+        return
+
+    exc.error = {}
 
+    if exc.args:
+        from apiclient import errors
+
+        http_error = exc.args[0]
+        assert isinstance(http_error, errors.HttpError)
```

Comment:
is it always HttpError? should we just ignore, keep it empty and raise to the user unknown exception

Reply:
HttpError currently is always passed to ApiRequestError. As discussed above, HttpError's json parsing is moved to PyDrive2 https://github.com/iterative/PyDrive2/pull/18